### PR TITLE
cmd/swarm: Add pss smoke test stub

### PR DESCRIPTION
**Pull request by nolash**
This PR adds a modest beginning on how a pss smoke test could be set up.

It is intended as a stub for anyone who feels like it to elaborate on.

### CODE REVIEW

File: cmd/swarm/swarm-smoke/pss.go
Comment (acud) [inline cmd/swarm/swarm-smoke/pss.go]: remove?

File: cmd/swarm/swarm-smoke/pss.go
Comment (acud) [inline cmd/swarm/swarm-smoke/pss.go]: not sure if this is necessary. you can just wrap `pssAsym` directly

File: cmd/swarm/swarm-smoke/pss.go
Comment (acud) [inline cmd/swarm/swarm-smoke/pss.go]: can we please have a brief text describing what we are testing, what are the test's erroneous stop conditions (if they at all exist), does the test have a green exit condition? or does it just go on running till it fails?

File: cmd/swarm/swarm-smoke/pss.go
Comment (acud) [inline cmd/swarm/swarm-smoke/pss.go]: remove?

File: cmd/swarm/swarm-smoke/pss.go
Comment (acud) [inline cmd/swarm/swarm-smoke/pss.go]: should we continue or fail the whole test?
what about metrics? I'd like to see a counter being incremented here

File: cmd/swarm/swarm-smoke/pss.go
Comment (acud) [inline cmd/swarm/swarm-smoke/pss.go]: remove code block?

File: cmd/swarm/swarm-smoke/pss.go
Comment (acud) [inline cmd/swarm/swarm-smoke/pss.go]: rm

**Review by acud — CHANGES_REQUESTED** [review]

File: cmd/swarm/swarm-smoke/pss.go
Comment (skylenet) [reply cmd/swarm/swarm-smoke/pss.go]: I've update the test to send multiple messages (configurable via CLI flag `pss-messages`) via PSS using asymetric encryption between random nodes.

I've added the following counters:

`pss.asym.failMsg` - total failed messages
`pss.asym.successMsg` - total successful messages

File: cmd/swarm/swarm-smoke/main.go
Comment (nonsense) [inline cmd/swarm/swarm-smoke/main.go]: asymmetric

File: cmd/swarm/swarm-smoke/pss.go
Comment (nonsense) [inline cmd/swarm/swarm-smoke/pss.go]: This should be structured log line - odd number of arguments, for example:
`log.Warn(fmt.Sprintf("message count should be positive. defaulting to %v", pssMessageCount))` in order to have 1 argument.

File: cmd/swarm/swarm-smoke/pss.go
Comment (nonsense) [inline cmd/swarm/swarm-smoke/pss.go]: comments in `godoc` start with the name of the method described.
`// pssAsymDo sends a single PSS message ...`

File: cmd/swarm/swarm-smoke/pss.go
Comment (nonsense) [inline cmd/swarm/swarm-smoke/pss.go]: probably better to do a while loop, although 20 times should be enough :)

File: cmd/swarm/swarm-smoke/pss.go
Comment (nonsense) [inline cmd/swarm/swarm-smoke/pss.go]: if you add a while loop above, this is not needed.

File: cmd/swarm/swarm-smoke/pss.go
Comment (nonsense) [inline cmd/swarm/swarm-smoke/pss.go]: you can just use `time.After(1*time.Second)` - it is simpler than context with a timeout and cancel function.

File: cmd/swarm/swarm-smoke/pss.go
Comment (nonsense) [inline cmd/swarm/swarm-smoke/pss.go]: maybe `log.Debug`

**Review by nonsense — APPROVED** [review]
lgtm provided this works as expected on k8s

**Comment by skylenet** [comment]
@justelad @nonsense thanks for the feedback. Let me know if there is anything else. Otherwise I'll open this on upstream.

Next steps will be done on separate PRs:
- Add grafana dashboards for these metrics
- Also use symmetrical encryption on the PSS smoke tests

**Comment by nonsense** [comment]
The `swarm-smoke` binary is starting to grow. We have `sync`, `pss` and `feed` tests in it, we should review it at some point and maybe think about splitting, before it gets too large so that we introduce bugs in it.

**Comment by skylenet** [comment]
> The `swarm-smoke` binary is starting to grow. We have `sync`, `pss` and `feed` tests in it, we should review it at some point and maybe think about splitting, before it gets too large so that we introduce bugs in it.

@nonsense I think this is a good idea. So I've split the `pss` tests already to a new binary.

File: cmd/swarm/swarm-smoke-pss/pss.go
Comment (nonsense) [inline cmd/swarm/swarm-smoke-pss/pss.go]: waitForMsg

File: cmd/swarm/swarm-smoke-pss/pss.go
Comment (nonsense) [inline cmd/swarm/swarm-smoke-pss/pss.go]: nitpick, but i introduced `tuid` as a `test uuid` in order to get all logs from a specific test. this is no longer needed, as i've noticed we can filter by the job id on elastic search, so `tuid` is completely redundant now. also it is not passed to the `runCheck` function.

feel free to remove or leave as is for now.

Comment (skylenet) [reply cmd/swarm/swarm-smoke-pss/pss.go]: I've removed it.

**Comment by skylenet** [comment]
Opened upstream https://github.com/ethereum/go-ethereum/pull/19338

**Pull request closed without merging**
